pkg/apis/build/v1alpha1/build_types_test.go

```diff
@@ -57,16 +59,57 @@ func TestBuildConditions(t *testing.T) {
 	}
 
 	// Add a new condition.
-	rev.Status.SetCondition(foo)
+	b.Status.SetCondition(foo)
+
+	if len(b.Status.Conditions) != 1 {
+		t.Fatalf("Unexpected Condition length; want 1, got %d", len(b.Status.Conditions))
+	}
+
+	foobuildCondition := b.Status.GetConditions()[0]
+	if cmp.Diff(foobuildCondition.Type, foo.Type) != "" {
+		t.Fatalf("Unexpected build condition type; want %v got %v", foo.Type, foobuildCondition.Type)
+	}
 
-	if len(rev.Status.Conditions) != 1 {
-		t.Fatalf("Unexpected Condition length; want 1, got %d", len(rev.Status.Conditions))
+	if cmp.Diff(foobuildCondition.Status, foo.Status) != "" {
+		t.Fatalf("Unexpected build condition status; want %v got %v", foo.Status, foobuildCondition.Type)
 	}
 
 	// Add a second condition.
-	rev.Status.SetCondition(bar)
+	b.Status.SetCondition(bar)
+
+	if len(b.Status.Conditions) != 2 {
```

Comment:
Can you `cmp.Diff` against the full expected conditions?

```
got := b.Status.Conditions()
want := []Condition{{
 // first
}, {
 // second
}}
if d := cmp.Diff(got, want); d != "" {
  t.Fatalf("got diff: %s", d)
}
```

(Printing the diff string is useful because it will include only exactly what changed, which can save a lot of time when debugging)

Reply:
I did try to compare `Condition` object directly but there is timestamp which causes `cmp.Diff` to fail. Thats why I went down the path of adding assertions for `Status` and `Type` individually